File: services/whatsapp-gateway/src/routes/health.ts
import { FastifyInstance } from "fastify";
import { db } from "../utils/db.js";
import { redisClient } from "../utils/redis.js";

export async function healthRoutes(app: FastifyInstance) {
  app.get("/health", async (_req, reply) => {
    const checks: Record<string, string> = {};

    // PostgreSQL
    try {
      await db.query("SELECT 1");
      checks.postgres = "ok";
    } catch {
      checks.postgres = "error";
    }

    // Redis
    try {
      await redisClient.ping();
      checks.redis = "ok";
    } catch {
      checks.redis = "error";
    }

    const healthy = Object.values(checks).every((v) => v === "ok");
    reply.status(healthy ? 200 : 503).send({
      status: healthy ? "healthy" : "degraded",
      timestamp: new Date().toISOString(),
      checks,
    });
  });

  app.get("/ready", async (_req, reply) => {
    reply.send({ status: "ready" });
  });
}

File: services/intelligence-service/src/routes/health.py
from fastapi import APIRouter
from src.utils.db import db_pool
from src.utils.redis_client import redis_pool

router = APIRouter()


@router.get("/health")
async def health():
    checks = {}
    try:
        await db_pool.fetchval("SELECT 1")
        checks["postgres"] = "ok"
    except Exception:
        checks["postgres"] = "error"

    try:
        await redis_pool.client.ping()
        checks["redis"] = "ok"
    except Exception:
        checks["redis"] = "error"

    healthy = all(v == "ok" for v in checks.values())
    return {"status": "healthy" if healthy else "degraded", "checks": checks}


@router.get("/ready")
async def ready():
    return {"status": "ready"}
